app/app.go

```diff
@@ -97,36 +97,36 @@ func (a *App) configureEventsForwarder() error {
 	if err != nil {
 		return err
 	}
-	a.server = NewServer(k, a.log, a.config)
+	sender := NewSender(k, a.log, a.config)
+	a.server = NewServer(sender, a.log)
 	return nil
 }
 
-// MetricsReporterInterceptor interceptor
-func (a *App) MetricsReporterInterceptor(
+// metricsReporterInterceptor interceptor
+func (a *App) metricsReporterInterceptor(
 	ctx context.Context,
 	req interface{},
 	info *grpc.UnaryServerInfo,
 	handler grpc.UnaryHandler,
 ) (interface{}, error) {
 
 	l := a.log.WithField("route", info.FullMethod)
-	ev := req.(*pb.Event)
 
 	startTime := time.Now()
 
 	defer func() {
 		timeUsed := float64(time.Since(startTime).Nanoseconds() / (1000 * 1000))
-		metrics.APIResponseTime.WithLabelValues(hostname, info.FullMethod, ev.GetTopic()).Observe(timeUsed)
+		metrics.APIResponseTime.WithLabelValues(hostname, info.FullMethod).Observe(timeUsed)
```

Comment:
Why are you removing the topic from the metric? Is it because now each event type will have different topics and we may have issues with the metrics cardinality?

Reply:
This part is WIP. I'll bring that back, was thinking if looping over all topics to increment metrics would be bad, but I don't think it is. This overhead would be paid (worse) with single events anyway.